diff -up ufw-0.36.20200125/setup.py.trans-dir ufw-0.36.20200125/setup.py
--- ufw-0.36.20200125/setup.py.trans-dir	2020-01-23 16:53:34.000000000 +0100
+++ ufw-0.36.20200125/setup.py	2020-01-25 23:53:28.236211046 +0100
@@ -61,6 +61,7 @@ class Install(_install, object):
             real_statedir = self.home + real_statedir
             real_prefix = self.home + '/usr'
         real_sharedir = os.path.join(real_prefix, 'share', 'ufw')
+        real_transdir = os.path.join(real_prefix, 'share', 'locale')
 
         # Update the modules' paths
         for fn in [ 'common.py' ]:
@@ -97,6 +98,11 @@ class Install(_install, object):
                                  "s%#SHARE_DIR#%" + real_sharedir + "%g",
                                  f])
 
+                subprocess.call(["sed",
+                                 "-i",
+                                 "s%#TRANS_DIR#%" + real_transdir + "%g",
+                                 f])
+
                 if fn == 'common.py' and 'UFW_SKIP_CHECKS' in os.environ and \
                    os.environ['UFW_SKIP_CHECKS'] != '':
                     print("Updating do_checks")
@@ -143,14 +149,20 @@ class Install(_install, object):
         self.copy_file('src/ufw-init-functions', init_helper_functions)
 
         # Install translations
-        transdir = real_sharedir
+        transdir = real_transdir
         if self.root != None:
-            transdir = self.root + real_sharedir
-        i18ndir = os.path.join(transdir, 'messages')
-        self.mkpath(i18ndir)
+            transdir = self.root + real_transdir
         if len(os.listdir('locales/mo')) == 0:
             subprocess.call(["make", "mo"])
-        self.copy_tree('locales/mo', i18ndir)
+        mo_files = os.listdir('locales/mo')
+        for f in mo_files:
+            if f.endswith('.mo'):
+                file_full_path = os.path.join('locales/mo', f)
+                lang = f[0:len(f)-3]
+                i18ndir = os.path.join(transdir, lang, 'LC_MESSAGES')
+                dest_full_path = os.path.join(i18ndir, 'ufw.mo')
+                self.mkpath(i18ndir)
+                self.copy_file(file_full_path, dest_full_path)
 
         # Install configuration files
         confdir = real_confdir
diff -up ufw-0.36.20200125/src/common.py.trans-dir ufw-0.36.20200125/src/common.py
--- ufw-0.36.20200125/src/common.py.trans-dir	2020-01-23 16:53:34.000000000 +0100
+++ ufw-0.36.20200125/src/common.py	2020-01-25 23:53:28.236211046 +0100
@@ -31,7 +31,7 @@ if False:
 programName = "ufw"
 state_dir = "#STATE_PREFIX#"
 share_dir = "#SHARE_DIR#"
-trans_dir = share_dir
+trans_dir = "#TRANS_DIR#"
 config_dir = "#CONFIG_PREFIX#"
 prefix_dir = "#PREFIX#"
 iptables_dir = "#IPTABLES_DIR#"
diff -up ufw-0.36.20200125/src/ufw.trans-dir ufw-0.36.20200125/src/ufw
--- ufw-0.36.20200125/src/ufw.trans-dir	2020-01-25 23:53:28.236211046 +0100
+++ ufw-0.36.20200125/src/ufw	2020-01-25 23:54:29.611211670 +0100
@@ -58,8 +58,7 @@ for i in sys.argv:
         args.append(i)
 
 # Internationalization
-gettext.bindtextdomain(domain, os.path.join(_findpath(ufw.common.trans_dir,
-                                                      rootdir), 'messages'))
+gettext.bindtextdomain(domain, _findpath(ufw.common.trans_dir, rootdir))
 try:
     # BAW: I'm not sure why both this and the .install() above is here, but
     # let's roll with it for now.  This is the Python 2 version, which ensures
diff -up ufw-0.36.20200125/tests/unit/support.py.trans-dir ufw-0.36.20200125/tests/unit/support.py
--- ufw-0.36.20200125/tests/unit/support.py.trans-dir	2020-01-23 16:53:34.000000000 +0100
+++ ufw-0.36.20200125/tests/unit/support.py	2020-01-25 23:53:28.236211046 +0100
@@ -72,7 +72,8 @@ def initvars(install_dir):
                             os.path.realpath(topdir), "ufw/lib/ufw")
     ufw.common.share_dir = os.path.join(
                             os.path.realpath(topdir), "ufw/usr/share/ufw")
-    ufw.common.trans_dir = ufw.common.share_dir
+    ufw.common.trans_dir = os.path.join(
+                            os.path.realpath(topdir), "ufw/usr/share/locale")
     ufw.common.prefix_dir = os.path.join(os.path.realpath(topdir), "ufw/usr")
 
 
